src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java

```diff
@@ -108,45 +109,56 @@ private void processQuery(JobContext context) throws IOException, InterruptedExc
     }
   }
 
-  private String generateQuery(String partitionFromDate, String partitionToDate, String project, String dataset,
-                               String table, @Nullable String serviceFilePath) {
-    if (partitionFromDate == null && partitionToDate == null) {
+  private String generateQuery(String partitionFromDate, String partitionToDate, String filter,
+                               String project, String dataset, String table, @Nullable String serviceFilePath) {
+    if (partitionFromDate == null && partitionToDate == null && filter == null) {
       return null;
     }
     String queryTemplate = "select * from %s where %s";
     com.google.cloud.bigquery.Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, table,
                                                                                 serviceFilePath);
     StandardTableDefinition tableDefinition = Objects.requireNonNull(sourceTable).getDefinition();
     TimePartitioning timePartitioning = tableDefinition.getTimePartitioning();
-    if (timePartitioning == null) {
+    if (timePartitioning == null && filter == null) {
       return null;
     }
     StringBuilder condition = new StringBuilder();
-    String columnName = timePartitioning.getField() != null ? timePartitioning.getField() : DEFAULT_COLUMN_NAME;
 
-    LegacySQLTypeName columnType = null;
-    if (!DEFAULT_COLUMN_NAME.equals(columnName)) {
-      columnType = tableDefinition.getSchema().getFields().get(columnName).getType();
-    }
+    if (timePartitioning != null) {
+      String columnName = timePartitioning.getField() != null ? timePartitioning.getField() : DEFAULT_COLUMN_NAME;
 
-    if (partitionFromDate != null) {
-      if (LegacySQLTypeName.DATE.equals(columnType)) {
-        condition.append("TIMESTAMP(").append(columnName).append(")");
-      } else {
-        condition.append(columnName);
+      LegacySQLTypeName columnType = null;
+      if (!DEFAULT_COLUMN_NAME.equals(columnName)) {
+        columnType = tableDefinition.getSchema().getFields().get(columnName).getType();
+      }
+
+      if (partitionFromDate != null) {
+        if (LegacySQLTypeName.DATE.equals(columnType)) {
```

Comment:
There's too much nesting in this method, it makes it difficult to understand. 

This part was existing logic, but it should be simplified so that it happens once when the columnType is fetched above instead of in multiple clauses.

```
columnType = ...
if (LegacySQLTypeName.DATE.equals(columnType)) {
  columnName = "TIMESTAMP(\"" + columnName + "\")";
}

```